code/modules/mob/living/burgerchat.dm

```diff
@@ -1,40 +1,69 @@
 // Thanks to Burger from Burgerstation for the foundation for this
 
-/* 
+/*
 BYOND Forum posts that helped me:
 http://www.byond.com/forum/post/1133166
 http://www.byond.com/forum/post/1072433
 http://www.byond.com/forum/post/940994
 http://www.byond.com/docs/ref/skinparams.html#Fonts
 */
 
+#define COLOR_JOB_UNKNOWN "#dda583"
+#define COLOR_PERSON_UNKNOWN "#999999"
+
+//For jobs that aren't roundstart but still need colours
+GLOBAL_LIST_INIT(job_colors_pastel, list(
+	"Prisoner" = 		"#d38a5c",
+	"CentCom" = 		"#90FD6D",
+	"Unknown"=			COLOR_JOB_UNKNOWN,
+))
+
 /mob/living
 	var/list/stored_chat_text = list()
 
 /proc/animate_chat(mob/living/target, message, message_language, message_mode, list/show_to, duration)
 
 	var/static/list/chatOverhead_colors = list("#83c0dd","#8396dd","#9983dd","#dd83b6","#dd8383","#83dddc","#83dd9f","#a5dd83","#ddd983","#dda583","#dd8383")
+	var/text_color
+
+	var/mob/living/carbon/human/target_as_human = target
+	if(istype(target_as_human))
+		if(target_as_human.wear_id?.GetID())
+			var/obj/item/card/id/idcard = target_as_human.wear_id
+			var/datum/job/wearer_job = SSjob.GetJob(idcard.GetJobName())
+			if(wearer_job)
+				text_color = wearer_job.chat_color
+			else
+				text_color = GLOB.job_colors_pastel[idcard.GetJobName()]
+		else
+			text_color = COLOR_PERSON_UNKNOWN
+	else
+		text_color = pick(chatOverhead_colors)
```

Comment:
Remove the list, it's redundant now and we can just use `COLOR_JOB_UNKNOWN`.

Reply:
I like the mobs like Ian having nicer say colours than hostile mobs, so I'm thinking of adding a var on mob that lets you choose the chat colour in case the mob is not of type /mob/living/carbon/human